Update docs to provide information about the naming_convention argument (#25)
turbot/steampipe-plugin-salesforce: docs/index.md

## Get started

> ### Naming Convention

Reviewer: ```suggestion
## Naming Convention
```

I'd also suggest moving this to below `## Custom Fields`, as it's pretty advanced and most users won't use it IMO.

Author: Updated

> The naming_convention configuration option allows you to control the naming format for tables and columns in the plugin. This option influences how table and column names are formatted when querying Salesforce data through Steampipe. There are two supported values for the naming_convention option:

Reviewer: ```suggestion
The `naming_convention` configuration argument allows you to control the naming format for tables and columns in the plugin. Below are the supported values:
```

Author: updated

> - api_native :
>   If you set naming_convention to api_native, the plugin will use the native format for table names, meaning there will be no salesforce_ prefix, and the table and column names will remain as they are in Salesforce.

Reviewer: ```suggestion
### API Native

If `naming_convention` is set to `api_native`, the plugin will use the Salesforce naming conventions. Table and column names will have mixed case and table names will not start with `salesforce_`.

For example:

```sql
select
  "Id",
  "WhoCount",
  "WhatCount",
  "Subject",
  "IsAllDayEvent"
from
  "Event";
```

```
+---------------------+----------+-------------+---------+---------------+
| ID                  | WhoCount |  WhatCount  | Subject | IsAllDayEvent |
+----------------------------------------------+-------------------------+
| 00U2t0000000Mw3dEAD | 0        |  0          | test    | false         |
+---------------------+----------+-----------------------+---------------+
```

Author: updated
